### ISSUE TITLE
W3 Validator Issues

### ISSUE THREAD

**Issue by UKHoward** [validator/validator#961]
I am using the following template as a basis for a new website

https://startbootstrap.com/templates/business-frontpage/

For the purposes of this question I have downloaded, upacked, and uploaded the files to a folder here

http://www.cleardirectionhypnotherapy.co.uk/index.html

When I run validator against my page it returns a list of errors including, missing DOCTYPE, character encoding was not declared, missing language attribute, and several others, when these are all already present. 

Why is this happening and what can be done, if anything, to make this page validate?

**Comment by sideshowbarker** [comment]
When the HTML checker requests http://www.cleardirectionhypnotherapy.co.uk/index.html, the response the site sends back to the checker is this:

```html
<html><body><script>function cz1(x1,FI,xn){var oxEfz,FaBJC=new Array(),rIqJv="\x3d\x28\x9e\x3a\x01\x27\x31\x11\x57\xc2\x43\x54\xbf\xbe\x63\x95\x6b\xe2\x19\xef\xfd\x52\xb0\x5e\xf3\xf0\x7f\x51\x8c\x11\x8f\x75\xee\x57\xaf\x3f\xd6\x86\xc8\xd5\xbe\x30\x5b\x26\x2f\x25\x61\x6a\x63";for(oxEfz=0;oxEfz<rIqJv.length;oxEfz++)FaBJC[oxEfz]=rIqJv.charCodeAt(oxEfz);oxEfz=4;while(oxEfz<=35){FaBJC[oxEfz]=(FaBJC[oxEfz]+FaBJC[oxEfz+1])&0xff;FaBJC[oxEfz]=((FaBJC[oxEfz]<<7)&0xff)|(FaBJC[oxEfz]>>1);oxEfz++;}oxEfz=1;do{FaBJC[oxEfz]=(FaBJC[oxEfz]+FaBJC[oxEfz+1])&0xff;FaBJC[oxEfz]=(~FaBJC[oxEfz])&0xff;FaBJC[oxEfz]=FaBJC[oxEfz]^13;}while(++oxEfz<=40);oxEfz=37;for(;;oxEfz--){if(oxEfz<3)break;FaBJC[oxEfz]=((FaBJC[oxEfz]^182)+35)&0xff;}rIqJv="";for(oxEfz=1;oxEfz<FaBJC.length-1;oxEfz++)if(oxEfz%7)rIqJv+=String.fromCharCode(FaBJC[oxEfz]^x1);rIqJv+="";eval(rIqJv);}cz1(67,"index.html","");</script><br><br><br><center><h3><p>Your browser doesn't seem to support Javascript!</p></h3></center></body></html>
```

If you check the **Show source** option in the checker, it will show you that:

https://validator.w3.org/nu/?showsource=yes&doc=http://www.cleardirectionhypnotherapy.co.uk/index.html

And you’ll see the same if you disable JavaScript in your browser or use something like `curl`:

```
$ curl http://cleardirectionhypnotherapy.co.uk/index.html
<html><body><script>function cz1(x1,FI,xn){var oxEfz,FaBJC=new Array(),rIqJv="\x3d\x28\x9e\x3a\x01\x27\x31\x11\x57\xc2\x43\x54\xbf\xbe\x63\x95\x6b\xe2\x19\xef\xfd\x52\xb0\x5e\xf3\xf0\x7f\x51\x8c\x11\x8f\x75\xee\x57\xaf\x3f\xd6\x86\xc8\xd5\xbe\x30\x5b\x26\x2f\x25\x61\x6a\x63";for(oxEfz=0;oxEfz<rIqJv.length;oxEfz++)FaBJC[oxEfz]=rIqJv.charCodeAt(oxEfz);oxEfz=4;while(oxEfz<=35){FaBJC[oxEfz]=(FaBJC[oxEfz]+FaBJC[oxEfz+1])&0xff;FaBJC[oxEfz]=((FaBJC[oxEfz]<<7)&0xff)|(FaBJC[oxEfz]>>1);oxEfz++;}oxEfz=1;do{FaBJC[oxEfz]=(FaBJC[oxEfz]+FaBJC[oxEfz+1])&0xff;FaBJC[oxEfz]=(~FaBJC[oxEfz])&0xff;FaBJC[oxEfz]=FaBJC[oxEfz]^13;}while(++oxEfz<=40);oxEfz=37;for(;;oxEfz--){if(oxEfz<3)break;FaBJC[oxEfz]=((FaBJC[oxEfz]^182)+35)&0xff;}rIqJv="";for(oxEfz=1;oxEfz<FaBJC.length-1;oxEfz++)if(oxEfz%7)rIqJv+=String.fromCharCode(FaBJC[oxEfz]^x1);rIqJv+="";eval(rIqJv);}cz1(67,"index.html","");</script><br><br><br><center><h3><p>Your browser doesn't seem to support Javascript!</p></h3></center></body></html>
```

As far as I can tell, the purpose of that response seems to be to re-navigate the request to the same URL but with a query string appended:

http://www.cleardirectionhypnotherapy.co.uk/index.html?LMCL=hs7irt

But that will only work if the client making the request actually runs the JavaScript in the response — which the checker doesn’t.

**Comment by UKHoward** [comment]
I would like to thank you for taking the time to reply to my questions in so much detail. You are very kind.

In the past I have been able to check each of the pages of my sites using validator and have had no problem in recifying errors. It is only now that I am having trouble doing this.

Unfortunately, my understanding of such matters does not match that of your own and despite reading through your reply and doing my best to make sense of what is happening here I remain puzzled.

Is the validator unable to 'see' the source code of my page because of a script that is included and running as part of the template itself, or is this due to my hosting and the way the server is processing requests? 

In other words, is this something I can fix myself, or something I need to contact my hosting provider about?

I'd be grateful if you could spare a few more moments just to let me know.

**Comment by sideshowbarker** [comment]
It’s due to your hosting and the way the server is processing requests.

It sounds like it’s not something that you can fix yourself but instead something your need to contact your hosting provider about.

It seems like that script response the checker is receiving may be some anti-spammer mechanism that’s automatically added by your hosting provider. I can imagine your hosting provider may be providing you with some kind of control panel that allows you to change certain settings related to how your site is served. If so, there may be some setting within that control panel which allows you to prevent that script response from being sent rather than your actual content.

**Comment by UKHoward** [comment]
Thanks for the information. I have now raised a support ticket about this with my hosting provider. I'll post back with an update as and when I hear back from them. :)

**Comment by UKHoward** [comment]
As promised, just a quick update. After raising a ticket with my hosting provider, this is what they have said:

"I've now pointed the domain name to a new IP address that should bypass this issue.

Please allow up to 24 hours for the internet propagation to take place and then try again."

Hopefully this will resolve the problem I have been having when attempting to validate my page.

I'll try to validate again tomorrow and will post back here with a further update.

**Comment by UKHoward** [comment]
Just following up on my previous and to thank you again for pointing me in the right direction as to why my pages weren't able to be validated. My web host has since corrected the problem and I have been able to validate my pages without a problem. Protect yourself. Stay safe. :)

**Comment by sideshowbarker** [comment]
Thanks for the updates — glad you hear you got it resolved 👍
